src/devices/Ina226/Ina226.cs

```diff
@@ -0,0 +1,489 @@
+        /// <param name="rShunt">Value of shunt in ohms. a 100amp 75mv shunt would be 0.00075 for example.</param>
+        /// <param name="maxExpectedCurrent">Maximum current (amps) expected when the voltage difference is 81.92mv, the INA226 full scale voltage.</param>
+        [Command]
+        public void SetCalibration(double rShunt, double maxExpectedCurrent)
+        {
+            _currentLsb = (maxExpectedCurrent / 3.2768e+4); // Math.Pow(2, 15) = 3.2768e+4
+            _calibrationValue = Convert.ToUInt16(0.00512 / (_currentLsb * rShunt)); // 0.00512 is given in the datasheet
+            WriteRegister(Ina226Register.Calibration, _calibrationValue);
+        }
+
+        /// <summary>
+        /// Function used to determine if a specific bit is set in a ushort register string
+        /// </summary>
+        /// <param name="bits">ushort read from a register on the INA226.</param>
+        /// <param name="pos">What position the bit to check is at.</param>
+        /// <returns>true if the bit is set and false if it is not.</returns>
+        internal static bool IsBitSet(ushort bits, int pos)
+        {
+            return (bits & (1 << pos)) != 0;
+        }
+
+        /// <summary>
+        /// Read the measured shunt voltage
+        /// </summary>
+        /// <returns>The shunt potential difference</returns>
+        [Telemetry("ShuntVoltage")]
+        public ElectricPotential ReadShuntVoltage()
+        {
+            ushort regValue = ReadRegister(Ina226Register.ShuntVoltage, _shuntConvTimeInt * _samplesAveragedInt);
+
+            if ((regValue & 0x8000) > 0)
+            {
+                regValue &= 0x7FFF;
+                regValue ^= 0x7FFF;
+                regValue++;
+
+                return ElectricPotential.FromMicrovolts(regValue * -10.0);
+            }
+
+            return ElectricPotential.FromMicrovolts(regValue * 10.0);
+        }
+
+        /// <summary>
+        /// Read the measured Bus Voltage
+        /// </summary>
+        /// <returns>The Bus potential (voltage)</returns>
+        [Telemetry("BusVoltage")]
+        public ElectricPotential ReadBusVoltage() => ElectricPotential.FromMillivolts((short)ReadRegister(Ina226Register.BusVoltage, _busConvTimeInt * _samplesAveragedInt) * 1.25);
+
+        /// <summary>
+        /// Read the calculated current through the INA226
+        /// </summary>
+        /// <remarks>
+        /// This value is determined by an internal calculation using the calibration register and the read shunt voltage and then scaled.
+        /// </remarks>
+        /// <returns>The calculated current</returns>
+        [Telemetry("Current")]
+        public ElectricCurrent ReadCurrent()
+        {
+            ushort regValue = ReadRegister(Ina226Register.Current, _shuntConvTimeInt * _samplesAveragedInt);
+            if ((regValue & 0x8000) > 0)
+            {
+                regValue &= 0x7FFF;
+                regValue ^= 0x7FFF;
+                regValue++;
```

Comment:
Just sign-extending would probably be easier to understand: 

int v = (int)(regValue | 0xFFFF0000);

Reply:
switched this out to make it easier to read/understand.